server/api/runtime_handlers/daskjob.py

```diff
@@ -397,11 +397,24 @@ def get_env_name(env_: Union[client.V1EnvVar, dict]) -> str:
         resources=spec.worker_resources, args=worker_args, **container_kwargs
     )
 
+    # We query the project to enrich the worker and scheduler pod spec with the project's default node selector.
+    # since the dask runtime operates locally and bypasses the usual server API flow, we do not enrich the run object.
+    # however, the node selector is still relevant for the Dask cluster's workers and scheduler,
+    # this ensures that the cluster pods follow the project's specified node selection.
```

Comment:
```suggestion
    # We query the project to enrich the worker and scheduler pod spec with the project's default node selector.
    # Since the dask runtime is a local run, and does not run in a dedicated k8s pod, node selectors for that run
    # are irrelevant, so we do not enrich the run object with the project node selector.
    # However, the node selector is still relevant for the Dask cluster's workers and scheduler, which do run
    # remotely on k8s. This ensures that the cluster pods follow the project's specified node selection.
```
I think the wording and terminology here is quite important, and we should make sure to make it super clear the business/functional reasoning for doing this and not only the technological limitations (bypassing the API)